samples/MauiEmbedding/MauiEmbedding/App.cs

```diff
@@ -98,28 +102,32 @@ private static void RegisterRoutes(IViewRegistry views, IRouteRegistry routes)
 	{
 		views.Register(
 			new ViewMap(ViewModel: typeof(ShellViewModel)),
-			new ViewMap<MainPage, MainViewModel>(),
-			new ViewMap<MauiControlsPage, MauiControlsViewModel>(),
+			new ViewMap<MainPage, MainViewModel>()
+#if MAUI_EMBEDDING
+			,new ViewMap<MauiControlsPage, MauiControlsViewModel>(),
 			new ViewMap<MauiEssentialsPage, MauiEssentialsViewModel>(),
-			//new ViewMap<TelerikControlsPage, TelerikControlsViewModel>(),
+			new ViewMap<TelerikControlsPage, TelerikControlsViewModel>(),
 			new ViewMap<MauiColorsPage, MauiColorsViewModel>(),
 			new ViewMap<EsriMapsPage, EsriMapsViewModel>(),
 			new ViewMap<ExternalLibPage, ExternalLibViewModel>(),
 			new ViewMap<MCTControlsPage, MCTControlsViewModel>()
+#endif
 		);
 
 		routes.Register(
 			new RouteMap("", View: views.FindByViewModel<ShellViewModel>(),
 				Nested: new RouteMap[]
 				{
 					new RouteMap("Main", View: views.FindByViewModel<MainViewModel>()),
-					new RouteMap(nameof(MauiControlsViewModel), View: views.FindByViewModel<MauiControlsViewModel>()),
-					new RouteMap(nameof(MCTControlsViewModel), View: views.FindByViewModel<MCTControlsViewModel>()),
-					new RouteMap(nameof(MauiEssentialsViewModel), View: views.FindByViewModel<MauiEssentialsViewModel>()),
-					new RouteMap(nameof(MauiColorsViewModel), View: views.FindByViewModel<MauiColorsViewModel>()),
-					new RouteMap(nameof(EsriMapsViewModel), View: views.FindByViewModel<EsriMapsViewModel>()),
-					new RouteMap(nameof(ExternalLibPage), View: views.FindByViewModel<ExternalLibViewModel>()),
-					//new RouteMap(nameof(TelerikControlsViewModel), View: views.FindByViewModel<TelerikControlsViewModel>()),
+#if MAUI_EMBEDDING
```

Comment:
This is not at all what we discussed. All of the registration in the user code should not care if Maui is supported or not.

Reply:
This is like this because there are third party controls, like maps, that don't play nice with net7.0. Everything else is as discussed